presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -517,9 +517,11 @@ public static HiveColumnStatistics fromMetastoreApiColumnStatistics(ColumnStatis
         }
         if (columnStatistics.getStatsData().isSetBooleanStats()) {
             BooleanColumnStatsData booleanStatsData = columnStatistics.getStatsData().getBooleanStats();
+            OptionalLong numFalse = fromMetastoreNumFalse(booleanStatsData.getNumFalses());
+            OptionalLong numTrue = (booleanStatsData.getNumFalses() == -1 ? OptionalLong.empty() : OptionalLong.of(booleanStatsData.getNumTrues()));
             return createBooleanColumnStatistics(
-                    booleanStatsData.isSetNumTrues() ? OptionalLong.of(booleanStatsData.getNumTrues()) : OptionalLong.empty(),
-                    booleanStatsData.isSetNumFalses() ? OptionalLong.of(booleanStatsData.getNumFalses()) : OptionalLong.empty(),
+                    booleanStatsData.isSetNumTrues() ? numTrue : OptionalLong.empty(),
+                    booleanStatsData.isSetNumFalses() ? numFalse : OptionalLong.empty(),
                     booleanStatsData.isSetNumNulls() ? fromMetastoreNullsCount(booleanStatsData.getNumNulls()) : OptionalLong.empty());
```

Comment:
To me the above is somehow overly complicated. I would inline `fromMetastoreNumFalse ` and write something like:
```
            OptionalLong falseCount = OptionalLong.empty();
            OptionalLong trueCount = OptionalLong.empty();
            if (booleanStatsData.isSetNumFalses() &&  booleanStatsData.isSetNumTrues() && booleanStatsData.getNumFalses() != -1) {
                // Impala 'COMPUTE STATS' write 1 as the numTrue and -1 as the numFalse
                falseCount = OptionalLong.of(booleanStatsData.getNumFalses());
                trueCount = OptionalLong.of(booleanStatsData.getNumTrues());
            }
            return createBooleanColumnStatistics(
                    trueCount,
                    falseCount,
                    booleanStatsData.isSetNumNulls() ? fromMetastoreNullsCount(booleanStatsData.getNumNulls()) : OptionalLong.empty());
```

Reply:
OK. I previously just try to keep style-consistency with method like `fromMetastoreNullsCount()`